src/main/java/org/folio/edge/sip2/repositories/CirculationRepository.java

```diff
@@ -77,30 +84,89 @@
               // this is probably not the permanent location
               // this might require a call to inventory
               .permanentLocation(
-                  resource.getResource() == null ? "Unknown"
+                  resource.getResource() == null ? UNKNOWN
                       : resource.getResource().getJsonObject("item",
                           new JsonObject()).getJsonObject("location",
-                              new JsonObject()).getString("name", "Unknown"))
+                              new JsonObject()).getString("name", UNKNOWN))
               .build()));
   }
 
-  private class CheckinRequestData implements IRequestData {
+  /**
+   * Perform a checkout.
+   *
+   * @param checkout the checkout domain object
+   * @return the checkout response domain object
+   */
+  public Future<CheckoutResponse> checkout(Checkout checkout, SessionData sessionData) {
+    final String institutionId = checkout.getInstitutionId();
+    final String patronIdentifier = checkout.getPatronIdentifier();
+    final String itemIdentifier = checkout.getItemIdentifier();
+
+    final JsonObject body = new JsonObject()
+        .put("itemBarcode", itemIdentifier)
+        .put("userBarcode", patronIdentifier)
+        .put("servicePointId", sessionData.getScLocation());
+
+    final Map<String, String> headers = new HashMap<>();
+    headers.put("accept", "application/json");
+
+    final CheckoutRequestData checkoutRequestData =
+        new CheckoutRequestData(body, headers, sessionData);
+    final Future<IResource> result = resourceProvider.createResource(checkoutRequestData);
+
+    return result
+        .otherwise(() -> null)
+        .compose(resource -> {
+          final ZonedDateTime dueDate;
+          // This is a mess. Need to clean this up. The problem here is that the checkout has
+          // already succeeded, but something could be wrong with the returned data. The odds of
+          // this are low, so we should be able to simplify this logic. The "dueDate" field is not
+          // required, so a loan could be missing one. I am not sure what that means for SIP2
+          // where the due date field is required. Setting it to null is probably wrong and will
+          // likely break the template when building the SIP2 response.
+          if (resource.getResource() != null) {
```

Comment:
I think it is highly unlikely. The problem is that "dueDate" is not required anywhere. I think if it is not in the call we make here, it will not likely be in the call to the loan directly. I think we should wait and see on this one. We can turn the mess into a function or refactor it into something else when we implement the missing (optional) fields post photo op.